### ISSUE TITLE
ImageDataGenerator with image and mask

### ISSUE THREAD

**Issue by aunusualman** [keras-team/keras#13175]
<em>Please make sure that this is a Bug or a Feature Request and provide all applicable information asked by the template.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.</em>  

**System information**  
- Have I written custom code (as opposed to using example directory):  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Linux Ubuntu 16.04
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.12
- Keras version:  2.1.6
- Python version:  3.6
- CUDA/cuDNN version:  10.0
- GPU model and memory:  12196MB

You can obtain the TensorFlow version with:  
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"  
You can obtain the Keras version with:  
python -c 'import keras as k; print(k.__version__)'  

**Describe the current behavior**  
 I try to do data augmentation with ImageDataGenerator.flow(),the data is a mask and it's pixel values are between 0 and 5.After augmentation,the saved images's pixel values are resaled to [0,255],but I do not want do that.So i check the Source code and find the problem. In image.py,function _get_batches_of_transformed_samples resacle the image.
**Describe the expected behavior**  
I hope it can save the original pixel values of the image.
**Code to reproduce the issue**  
    def _get_batches_of_transformed_samples(self, index_array):
        batch_x = np.zeros(tuple([len(index_array)] + list(self.x.shape)[1:]),
                           dtype=K.floatx())
        for i, j in enumerate(index_array):
            x = self.x[j]
            x = self.image_data_generator.random_transform(x.astype(K.floatx()))
            x = self.image_data_generator.standardize(x)
            batch_x[i] = x
        if self.save_to_dir:
            for i, j in enumerate(index_array):
                img = array_to_img(batch_x[i], self.data_format, scale=True)
                fname = '{prefix}_{index}_{hash}.{format}'.format(prefix=self.save_prefix,
                                                                  index=j,
                                                                  hash=np.random.randint(1e4),
                                                                  format=self.save_format)

**Other info / logs**  
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.
